source/s3_auto_ranged_put.c

```diff
@@ -86,8 +255,22 @@ struct aws_s3_meta_request *aws_s3_meta_request_auto_ranged_put_new(
         &auto_ranged_put->synced_data.etag_list, etag_c_array, num_parts, sizeof(struct aws_string *));
     auto_ranged_put->content_length = content_length;
     auto_ranged_put->synced_data.total_num_parts = num_parts;
+
     auto_ranged_put->threaded_update_data.next_part_number = 1;
 
+    if (options->resume_token) {
+        int load_persistable_state_error = s_load_persistable_state(
+            allocator, auto_ranged_put, content_length, multipart_upload_id, part_size, num_parts);
+        if (load_persistable_state_error) {
+            aws_raise_error(load_persistable_state_error);
+            goto error_clean_up;
```

Comment:
memory leak: `multipart_upload_id` needs to be destroyed.
Maybe clean it up from `error_clean_up`